request_logging/middleware.py

```diff
@@ -153,24 +155,28 @@ def _log_request(self, request):
         logging_context = self._get_logging_context(request, None)
         self.logger.log(logging.INFO, method_path, logging_context)
         self._log_request_headers(request, logging_context)
-        self._log_request_body(request, logging_context)
 
     def _log_request_headers(self, request, logging_context):
         headers = {k: v for k, v in request.META.items() if k.startswith('HTTP_')}
 
         if headers:
             self.logger.log(self.log_level, headers, logging_context)
 
-    def _log_request_body(self, request, logging_context):
+    def process_body(self, request):
         if request.body:
-            content_type = request.META.get('CONTENT_TYPE', '')
-            is_multipart = content_type.startswith('multipart/form-data')
-            if is_multipart:
-                self.boundary = '--' + content_type[30:]  # First 30 characters are "multipart/form-data; boundary="
-            if is_multipart:
-                self._log_multipart(self._chunked_to_max(request.body), logging_context)
-            else:
-                self.logger.log(self.log_level, self._chunked_to_max(request.body), logging_context)
+            if not self._should_log_route(request):
```

Comment:
This is too deeply nested for me. Let's do a quick return at the start for `if not request.body` and then break up the logic below into functions that handle the individual request types. The only real difference between the GET and POST is how we assemble what needs to be logged, so the functions should return that and then the logging happen inside here, whereas multipart is a little more complex; you could vary the logging method inside the `if multipart` check.

Reply:
To be clear, this all looks good in general and thanks for the contribution (I also realize a fair bit of the nesting was pre-existing).

Reply:
Ok sounds good, I'll play around with this tonight

Reply:
Updated, if you have any more suggestions or things you'd like reworked, let me know.